include/seqan/align/dp_formula_affine.h

```diff
@@ -95,12 +96,9 @@ _internalComputeScore(DPCell_<TScoreValue, TAffineGaps> & activeCell,
                       TracebackOn<TracebackConfig_<SingleTrace, TGapsPlacement> > const &,
                       RecursionDirectionDiagonal const &)
 {
-    if(activeCell._score <= rightCompare)
-    {
-        activeCell._score = rightCompare;
-        return TraceBitMap_<TScoreValue>::DIAGONAL | leftTrace;
-    }
-    return leftTrace | gapTrace;
+    return (activeCell._score <= rightCompare) ?
```

Comment:
Because, here we check the Diagonal direction which should always be preferred over horizontal or vertical direction if the score is the same.